### PR TITLE
Fixed update checker making requests to uptime.kuma.pet even when turned off

### PR DESCRIPTION
**Pull request by tofran**
> ⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have [read pull request rules](https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma).
> 
> Tick the checkbox if you understand [x]: 

- [x] I have read and understand the pull request rules.

# Description

Fixes the issue outlined in #1743

- Fixed **bug where uptime-kuma it would make the request to `uptime.kuma.pet` regardless of the `checkUpdate` config**;
- Defined constants in the top of the document for easier configuration/documentation;
- Removed unnecessary `compareVersions`: we were comparing the same var on both sides `res.data.beta`, so it will always be equal.

## Type of change

Bug fix.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [x] My code needed automated testing. I have added them (this is optional task)

### CODE REVIEW

File: server/check-version.js [hunk 1]
```diff
@@ -1,46 +1,49 @@
 const { setSetting, setting } = require("./util-server");
 const axios = require("axios");
-const compareVersions = require("compare-versions");
+const { log } = require("../src/util");
 
 exports.version = require("../package.json").version;
 exports.latestVersion = null;
 
+const UPDATE_CHECKER_INTERVAL_MS = 3600 * 1000 * 48;
```
Comment (Saibamen) [inline server/check-version.js:8]: Please add comment what is that

File: server/check-version.js [hunk 2]
```diff
@@ -1,46 +1,49 @@
...
 let interval;
 
-/** Start 48 hour check interval */
 exports.startInterval = () => {
     let check = async () => {
+        if (await setting("checkUpdate") === false) {
+            return;
+        }
+
+        log.debug("update-checker", "retrieving latest versions");
```
Comment (Saibamen) [inline server/check-version.js:19]:
```suggestion
        log.debug("update-checker", "Retrieving latest versions");
```

File: server/check-version.js [hunk 3]
```diff
@@ -1,46 +1,50 @@
 const { setSetting, setting } = require("./util-server");
 const axios = require("axios");
-const compareVersions = require("compare-versions");
+const { log } = require("../src/util");
 
 exports.version = require("../package.json").version;
 exports.latestVersion = null;
 
+// How much time in ms to wait between update checks
```
Comment (Saibamen) [inline server/check-version.js:8]: Add info about current value - 48h

**Review by Saibamen — CHANGES_REQUESTED** [review]

File: server/check-version.js (hunk 3 above)
Comment (tofran) [reply server/check-version.js:8]: I can do it, but I would advise against this. One should not create redundancy with comments, redundancy is likely to become out of sync with time.
Would changing `3600 * 1000 * 48` to `1000 * 60 * 60  * 48` help?
Should I proceed duplicating the current value to the comment? Is this the standard for this project?

**Comment by louislam** [comment]
I kept the compareVersions as it is not unnecessary, but it was fixed.

**Comment by tofran** [comment]
Thank you @louislam

**Pull request merged**